org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MavenPluginActivator.java

```diff
@@ -390,19 +255,19 @@ public static MavenPluginActivator getDefault() {
   }
 
   public MavenModelManager getMavenModelManager() {
-    return this.modelManager;
+    return this.mavenModelManagerTracker.getService();
```

Comment:
If we have refactored the whole m2e code yes, but I don't want to do to much at once. We even need to refactor the test as well.

Reply:
I have now cerated a dedicated getService method so we do not need all those fields, this will make removal also easier as we simply need to remove the specific method.

Reply:
> If we have refactored the whole m2e code yes, but I don't want to do to much at once. We even need to refactor the test as well.

That's reasonable. Once the the (de-facti) static methods are gone I think we can also get rid of the static `MavenPluginActivator` instance and maybe even the entire activator (which I assume is one goal).

> I have now cerated a dedicated getService method so we do not need all those fields, this will make removal also easier as we simply need to remove the specific method.

That's IMHO far better! Looks nice.